toolkits/google/arcade_google/tools/gmail.py

```diff
@@ -124,36 +185,85 @@ async def write_draft_email(
     Compose a new email draft using the Gmail API.
     """
     # Set up the Gmail API client
-    service = build(
-        "gmail",
-        "v1",
-        credentials=Credentials(
-            context.authorization.token
-            if context.authorization and context.authorization.token
-            else ""
-        ),
-    )
-
-    message = MIMEText(body)
-    message["to"] = recipient
-    message["subject"] = subject
-    if cc:
-        message["Cc"] = ", ".join(cc)
-    if bcc:
-        message["Bcc"] = ", ".join(bcc)
+    service = _build_gmail_service(context)
```

Comment:
I'm experiencing issues with this tool. Seems to not be working anymore.

```
[17:09:31.270] call_h9aNhdwdD4n3wHHoJy1zJ4yq | Tool developer message: Error in write_draft_email: <HttpError 400 when requesting https://gmail.googleapis.com/gmail/v1/users/me/drafts?alt=json returned "Missing draft message". Details: "[{'message': 'Missing draft message', 'domain': 'global', 'reason': 'invalidArgument'}]"> arcade.worker.core.base:call_tool:150
```

```
Tool inputs: {'body': 'Hi John,\n\nI hope this message finds you well. \n\nBest regards,\n\n[Your Name]', 'recipient': 'johndoe@arcade.dev', 'subject': 'Best wishes'}
```

Reply:
yep, need to use `MIMEText` for drafts and `EmailMessage` only when sending the email, was using the former for both actions. just pushed a fix for it